lib/unhangout-sockets.js

```diff
@@ -167,6 +167,17 @@ _.extend(UnhangoutSocketManager.prototype, events.EventEmitter.prototype, {
             }
         }, this));
 
+        mgr.on("store-contact", _.bind(function(socket, args) {
+            var event = this.getEvent(args.roomId);
+
+            var user = socket.user; 
+            user.set("preferredContact", args.preferredContact);
+            user.set("noShare", args.noShareVal);
+
+            user.save();
+
+        }, this));
```

Comment:
To maintain the convention with other socket calls, this function should send ``mgr.writeAck("store-contact")`` on success and ``mgr.writeErr("store-contact")`` on error.